### PR TITLE
fix freezing of backbones in classifier training. Add store labels and embeddings.

### PR DESCRIPTION
**Pull request by RaphaelKreft**

### CODE REVIEW

File: models/main_model_dlproj.py [hunk 1]
```diff
@@ -32,55 +32,67 @@ def __init__(self, config):
...
         self.switch_mode(self.training_mode)
 
 
+    def activate_train_mode(self):
+        self.backbone.train()
+        if self.classifier is not None:
+            self.classifier.train()
+
+        if self.training_mode == 'train-classifier':
+            self.freeze_backbone()
```
Comment (Shaswat-G) [inline models/main_model_dlproj.py:53]: I'm just curious: will this work appropriately for all training modes? I am assuming this function will not be called for classification and gen_embeddings.

1) We need to ensure that for all pre-training modes, the backbone is train mode and that self.classifier is None in these modes.
2) For mode = 'train-classifier', we freeze the backbone and assume that the classifier was put into training mode in the previous if block.

If my understanding is correct, it could be clearer to do this:

`


    def activate_train_mode(self):
        self.backbone.train()
        if self.training_mode == 'train-classifier':
            self.freeze_backbone()
             self.classifier.train()
`

**Review by Ahwodafo — APPROVED** [review]

File: models/main_model_dlproj.py (hunk 1 above)
Comment (Shaswat-G) [reply models/main_model_dlproj.py:53]: We can put assert statments for more clarity

File: train_crl_dlproj.py [hunk 2]
```diff
@@ -400,17 +411,16 @@ def test(train_bb=False, gen_embed=False, train_classifier=False, benchmark_clas
             "mode": "pretrain",
             "loss": "NTXent",
             "max_epochs": 2,
-            "batch_size": 16,
+            "batch_size": 128,
             "lr": 0.0005,
             "weight_decay": 0.0001,
             "temperature": 0.07,
-            "val_period": 649,
+            "val_period": 325,
```
Comment (Shaswat-G) [inline train_crl_dlproj.py:418]: Should we exclude these config changes since these are for testing only?

File: models/main_model_dlproj.py (hunk 1 above)
Comment (RaphaelKreft) [reply models/main_model_dlproj.py:53]: technically its not needed. (Classifier is only in train mode when train-classifier but activate_train_mode is only called in training tasks)

File: train_crl_dlproj.py (hunk 2 above)
Comment (RaphaelKreft) [reply train_crl_dlproj.py:418]: I would leave them there, does not hurt :)
